src/suricata-plugin.h

```diff
@@ -47,6 +47,7 @@ typedef struct SCPlugin_ {
     const char *license;
     const char *author;
     void (*Init)(void);
+    void (*RegisterLoggers)(void);
```

Comment:
Undecided if this should be a function pointer on this struct.  There could be other lifecycle callbacks we want as well.

Would it make more sense to provide functions to register for lifecycle callbacks?  For example, The plugin, in `Init()` could do:

```c
SCOnLoggerRegistration(MyRegisterLoggersCallback)
```

Then Suricata would call `MyRegisterLoggersCallback` when its ready for all types of loggers to be registered.